flag.go

```diff
@@ -595,3 +596,27 @@ func withEnvHint(envVar, str string) string {
 	}
 	return str + envText
 }
+
+func stringifyFlag(f Flag) string {
+	fv := reflect.ValueOf(f)
+	placeholder, usage := unquoteUsage(fv.FieldByName("Usage").String())
```

Comment:
I'm not crazy about the implicit contract here -- e.g. considering if a user wants to create their own flag. Is it possible to add the needed methods to the `Flag` interface to extract this data?